adi/adar1000.py

```diff
@@ -50,150 +69,121 @@ def __get__(self, instance, owner):
         )
 
     def __set__(self, instance, value):
-        instance._set_iio_attr_float(
-            self._channel_name, self._attr, self._output, value, self._dev
-        )
-
-
-def _add_prop(classname, channelname, attr, i, output, dev, beam_name=None):
-    """ Add dynamic property """
-    inoutstr = "tx" if output else "rx"
-    pn = inoutstr + str(i) + "_" + attr
-    if beam_name:
-        pn += "_" + beam_name.lower()
-
-    setattr(
-        classname,
-        pn,
-        _dyn_property_float(
-            attr=attr, dev=dev, channel_name=channelname, output=output
-        ),
-    )
+        try:
+            instance._set_iio_attr_float(
+                self._channel_name, self._attr, self._output, float(value), self._dev
+            )
+        except OSError:
+            instance._set_iio_attr_int(
+                self._channel_name, self._attr, self._output, int(value), self._dev
+            )
+        except ValueError:
+            instance._set_iio_attr(
+                self._channel_name, self._attr, self._output, str(value), self._dev
+            )
 
 
 class adar1000(attribute, context_manager):
-    """ ADAR1000 Beamformer
+    """ADAR1000 Beamformer
 
     parameters:
         uri: type=string
             URI of IIO context with ADAR1000(s)
-        beams: type=string,list[string]
-            String or list of strings identifying desired chip select
-            option of ADAR1000. This is based on the jumper configuration
-            if the EVAL-ADAR1000 is used. These strings are the labels
-            coinciding with each chip select and are typically in the
-            form BEAM0, BEAM1, BEAM2, BEAM3. Use a list when multiple
-            are chips are cascaded together. Dynamic class properties
-            will be created for each beam and signal path.
+        chip_ids: type=string,list[string]
+            String or list of strings identifying desired chip select and
+            hardware ID for the ADAR1000. These strings are the labels
+            coinciding with each chip select and hardware address and are
+            typically in the form csbX_chipX. The csb line can be any number
+            depending on how many are used in the system. The chip number
+            will typically be 1-4 because each CSB line can control up to
+            four ADAR1000s. Use a list when multiple chips are to be
+            controlled. Dynamic class properties will be created for each
+            ADAR1000 device.
     """
 
     _device_name = ""
-    _beam_channels = ["voltage0", "voltage1", "voltage2", "voltage3"]
 
-    def __init__(self, uri="", beams="BEAM0"):
+    def __init__(self, uri="", chip_ids="csb1_chip1"):
```

Comment:
Does this value map to the default driver name for ADAR1000? or is the specific to stingray?

Reply:
I'm going to recommend that the default is updated to this style since it's more explicit and easier to understand at a glance. The "BEAM" notation doesn't tell you anything about how to access the chip whereas this notation tells you everything about how to interface with that particular IC.